src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/TreeView/TreeViewPage.xaml.cs

```diff
@@ -0,0 +1,1010 @@
+		{
+			if (IsInContentMode())
+			{
+				SetAutomationIdForNodes(ContentModeTestTreeView);
+			}
+			else
+			{
+				SetAutomationIdForNodes(TestTreeView);
+			}
+		}
+
+		private void SetAutomationIdForNodes(TreeView tree)
+		{
+			var listControl = VisualTreeUtils.FindVisualChildByName(this.TestTreeView, "ListControl") as TreeViewList;
+			Stack<TreeViewNode> pendingNodes = new Stack<TreeViewNode>();
+			foreach (var node in tree.RootNodes)
+			{
+				pendingNodes.Push(node);
+			}
+			while (pendingNodes.Count > 0)
+			{
+				var currentNode = pendingNodes.Pop();
+				foreach (var child in currentNode.Children)
+				{
+					pendingNodes.Push(child);
+				}
+
+				var container = tree.ContainerFromNode(currentNode);
+				if (container != null)
+				{
+					AutomationProperties.SetAutomationId(container as DependencyObject, GetNodeContent(currentNode));
+				}
+			}
+		}
+
+		private String GetNodeContent(TreeViewNode node)
+		{
+			if (IsInContentMode())
+			{
+				TreeViewItemSource item = node.Content as TreeViewItemSource;
+				return item.Content;
+			}
+			return node.Content.ToString();
+		}
+
+		private string GetRootNodeChildrenOrder(TreeView tree)
+		{
+			List<string> result = new List<string>();
+			Stack<TreeViewNode> pendingNodes = new Stack<TreeViewNode>();
+			pendingNodes.Push(tree.RootNodes[0]);
+			while (pendingNodes.Count > 0)
+			{
+				var currentNode = pendingNodes.Pop();
+				var size = currentNode.Children.Count;
+				for (int i = 0; i < size; i++)
+				{
+					pendingNodes.Push(currentNode.Children[size - 1 - i]);
+				}
+
+				result.Add(GetNodeContent(currentNode));
+			}
+
+			return string.Join(" | ", result);
+		}
+
+		private string GetItemsSourceOrder()
+		{
+			List<string> result = new List<string>();
+			Stack<TreeViewItemSource> pendingItems = new Stack<TreeViewItemSource>();
+			pendingItems.Push(TestTreeViewItemsSource[0]);
+			while (pendingItems.Count > 0)
+			{
+				var currentItem = pendingItems.Pop();
+				var size = currentItem.Children.Count;
+				for (int i = 0; i < size; i++)
+				{
+					pendingItems.Push(currentItem.Children[size - 1 - i]);
+				}
+				result.Add(currentItem.Content);
+			}
+
+			return string.Join(" | ", result);
+		}
+
+		private void GetChildrenOrder_Click(object sender, RoutedEventArgs e)
+		{
+			if (IsInContentMode())
+			{
+				var itemsSourceOrder = GetItemsSourceOrder();
+				var treeViewNodeOrder = GetRootNodeChildrenOrder(ContentModeTestTreeView);
+				// Make sure ItemsSource and TreeViewNode orders are in sync
+				if (itemsSourceOrder == treeViewNodeOrder)
+				{
+					Results.Text = itemsSourceOrder;
+				}
+				else
+				{
+					Results.Text = $"ItemsSourceOrder: {itemsSourceOrder}; TreeViewNodeOrder: {treeViewNodeOrder}";
+				}
+			}
+			else
+			{
+				Results.Text = GetRootNodeChildrenOrder(TestTreeView);
+			}
+		}
+
+		private void ToggleSelectionMode_Click(object sender, RoutedEventArgs e)
+		{
+			var selectionMode = TreeViewSelectionMode.Single;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'selectionMode'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)